File: backend/app/schemas/page.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class PageCreate(BaseModel):
    title: str
    content_markdown: str = ""
    parent_id: int | None = None
    status: str = "draft"
    position: int = 0


class PageUpdate(BaseModel):
    title: str | None = None
    content_markdown: str | None = None
    parent_id: int | None = None
    status: str | None = None
    position: int | None = None
    change_message: str | None = None


class PageAutosaveRequest(BaseModel):
    title: str | None = None
    content_markdown: str | None = None


class PagePublishRequest(BaseModel):
    change_message: str | None = None


class PageCreateVersionRequest(BaseModel):
    change_message: str | None = None


class PageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    workspace_id: int
    parent_id: int | None
    title: str
    slug: str
    content_markdown: str
    status: str
    position: int
    created_by_id: int
    updated_by_id: int | None
    archived_at: datetime | None
    created_at: datetime
    updated_at: datetime


class PageListItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    workspace_id: int
    parent_id: int | None
    title: str
    slug: str
    status: str
    position: int
    updated_at: datetime


class PageVersionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    page_id: int
    version_number: int
    title: str
    content_markdown: str
    status: str
    change_message: str | None
    created_by_id: int
    created_at: datetime

File: frontend/src/types/page.ts
export type PageStatus = 'draft' | 'published' | 'archived';

export interface Page {
  id: number;
  workspace_id: number;
  parent_id: number | null;
  title: string;
  slug: string;
  content_markdown: string;
  status: PageStatus;
  position: number;
  created_by_id: number;
  updated_by_id: number;
  archived_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface PageListItem {
  id: number;
  workspace_id: number;
  parent_id: number | null;
  title: string;
  slug: string;
  status: PageStatus;
  position: number;
  updated_at: string;
}

export interface PageVersion {
  id: number;
  page_id: number;
  version_number: number;
  title: string;
  content_markdown: string;
  status: PageStatus;
  change_message: string | null;
  created_by_id: number;
  created_at: string;
}

export interface CreatePageBody {
  title: string;
  content_markdown?: string;
  parent_id?: number | null;
  status?: PageStatus;
  position?: number;
}

export interface UpdatePageBody {
  title?: string | null;
  content_markdown?: string | null;
  parent_id?: number | null;
  status?: PageStatus | null;
  position?: number | null;
  change_message?: string | null;
}

export interface PageAutosaveBody {
  title?: string | null;
  content_markdown?: string | null;
}

export interface CreatePageVersionBody {
  change_message?: string | null;
}

export interface PagePublishBody {
  change_message?: string | null;
}

export interface PageTreeNode extends PageListItem {
  children: PageTreeNode[];
}
